invoices/utils.py

```diff
@@ -136,18 +144,24 @@ def get_invoices():
         invoices_data[invoice_key] = invoice
     return invoices_data
 
+def get_weekly_reports():
+    weekly_report_data = {}
+    for weekly_report in WeeklyReport.objects.all():
```

Comment:
Wait, why are you fetching all weekly reports, instead of what is actually necessary?

Reply:
This works similar to get_invoices(). Shouldn't this work the same way, i.e. fetch all weekly reports from DB? What do you mean by "what is actually necessary"?

Reply:
You're right, `get_invoices` has the same thing. There is no need to fetch all the invoices/weekly reports when updating a specific time range, but maybe it is so much simpler it should stay that way.